specs/tree-view/tree-view.md

```diff
@@ -0,0 +1,195 @@
+
+Flat tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+    <fast-tree-item>Root node four</fast-tree-item>
+</fast-tree-view>
+```
+
+Nested tree:
+```html
+<fast-tree-view>
+    <fast-tree-item>Root node one</fast-tree-item>
+    <fast-tree-item>Root node two</fast-tree-item>
+    <fast-tree-item>
+        <fast-tree-item>Nested node one</fast-tree-item>
+        <fast-tree-item>Nested node two</fast-tree-item>
+    </fast-tree-item>
+    <fast-tree-item>Root node three</fast-tree-item>
+</fast-tree-view>
+```
+
+**Tree view item**
+
+Default
+![](./images/tree-item.png)
+
+Selected
+![](./images/tree-item-selected.png)
+
+Expanded
+![](./images/tree-item-expanded.png)
+
+Collapsed
+![](./images/tree-item-collapsed.png)
+
+Like tree view, the tree view item has a default slot for its children.
+
+Slots:
+- `content`
+- `before-content`
+- `after-content` 
+- `expand-collapse-glyph`
+
+Parts:
+- root
+- content
+- before-content
+- after-content
+- expand-collapse-glyph
+- child-nodes
+- expanded
+- selected
+
+---
+
+## Implementation
+
+*Important aspects of the planned implementation with careful consideration of web standards and integration.*
+
+### States
+
+**Tree View Item**
+
+Expanded:
+
+ A parent node with child nodes can be either expanded or collapsed (not expanded). The following state transitions or interactions are all valid:
+
+ - the control may be initialized with no parent nodes expanded
+ - the control may be initialized with all or some parent nodes expanded
+ - clicking on the expand/collapse button will expand or collapse a parent node
+ - in a vertical tree, when a parent node has focus, right arrow expands and left arrow collapses the parent node.
+ - in a vertical tree, when a parent node has focus, space bar expands and collapses the tree.
+
+Selected:
+
+In a single select tree, only one item may be selected at a time. Selection should not follow focus. The following state transitions or interactions are all valid:
+
+- if none of the nodes are selected before the tree receives focus, focus is set on the first node.
+- if a node is selected before the tree receives focus, focus is set on the selected node.
+- clicking on a node selects it
+- clicking on the expand/collapse button selects and performs the expand/collapse behavior
+- when focus is on a node, enter selects the node
+
+### Accessibility
+
+The tree view should align to the interaction model provided by the W3C: https://w3c.github.io/aria-practices/#TreeView
+
+### Globalization
+
+The component visuals should support RTL scenarios (per below):
+![](./images/tree-view-rtl.png)
+
+### Test Plan
+
+While testing is still TBD for our web components, I would expect this to align with the testing strategy and not require any additional test support.
```

Comment:
I think we should determine this and write a document for component testing we can just link in this section for components.

Reply:
I'm on point for writing the testing spec - we can update and link later when it goes in I think, yeah?